build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/NewRestTestBasePlugin.java

```diff
@@ -0,0 +1,224 @@
+    private static final String MODULES_CONFIGURATION = "clusterModules";
+    private static final String PLUGINS_CONFIGURATION = "clusterPlugins";
+    private static final String EXTRACTED_PLUGINS_CONFIGURATION = "extractedPlugins";
+
+    private final ProviderFactory providerFactory;
+
+    @Inject
+    public NewRestTestBasePlugin(ProviderFactory providerFactory) {
+        this.providerFactory = providerFactory;
+    }
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().apply(ElasticsearchJavaPlugin.class);
+        project.getPluginManager().apply(InternalDistributionDownloadPlugin.class);
+
+        // Register integ-test and default distributions
+        NamedDomainObjectContainer<ElasticsearchDistribution> distributions = DistributionDownloadPlugin.getContainer(project);
+        ElasticsearchDistribution defaultDistro = distributions.create(DEFAULT_REST_INTEG_TEST_DISTRO, distro -> {
+            distro.setVersion(VersionProperties.getElasticsearch());
+            distro.setArchitecture(Architecture.current());
+        });
+        ElasticsearchDistribution integTestDistro = distributions.create(INTEG_TEST_REST_INTEG_TEST_DISTRO, distro -> {
+            distro.setVersion(VersionProperties.getElasticsearch());
+            distro.setArchitecture(Architecture.current());
+            distro.setType(ElasticsearchDistributionTypes.INTEG_TEST_ZIP);
+        });
+
+        // Create configures for module and plugin dependencies
+        Configuration modulesConfiguration = createPluginConfiguration(project, MODULES_CONFIGURATION, true);
+        Configuration pluginsConfiguration = createPluginConfiguration(project, PLUGINS_CONFIGURATION, false);
+        Configuration extractedPluginsConfiguration = createPluginConfiguration(project, EXTRACTED_PLUGINS_CONFIGURATION, true);
+        extractedPluginsConfiguration.extendsFrom(pluginsConfiguration);
+        configureArtifactTransforms(project);
+
+        // For plugin and module projects, register the current project plugin bundle as a dependency
+        project.getPluginManager().withPlugin("elasticsearch.esplugin", plugin -> {
+            if (GradleUtils.isModuleProject(project.getPath())) {
+                project.getDependencies()
+                    .add(modulesConfiguration.getName(), project.getDependencies().project(Map.of("path", project.getPath())));
+            } else {
+                project.getDependencies().add(pluginsConfiguration.getName(), project.files(project.getTasks().named("bundlePlugin")));
+            }
+
+        });
+
+        project.getTasks().withType(StandaloneRestIntegTestTask.class, task -> {
+            SystemPropertyCommandLineArgumentProvider nonInputSystemProperties = task.getExtensions()
+                .getByType(SystemPropertyCommandLineArgumentProvider.class);
+
+            task.dependsOn(integTestDistro, modulesConfiguration);
+            registerDistributionInputs(task, integTestDistro);
```

Comment:
no mater if we call `usesDefaultDistribution` or not this will always stay the input so we'd build both distributions. we should make this lazy. We probably could use a NamedObjectProvider<ElasticsearchDistribution> here for that an materialize this in a provider by using `distributions.register` above

Reply:
I didn't think the complexity was worth while. The default distribution is a superset of the integtest one anyhow so you basically have to build all the same stuff.

Reply:
One other thing is that it then makes it impossible to author tests in a single project that use _both_ distributions, which is likely as we slowly start to migrate tests from requiring the default distro to using the integ-test distro. Unless we additionally add a DSL to register them independently.

Realistically the goal is to require the default distro less and less and use the integ-test distro exclusively.